Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -1485,6 +1638,32 @@ namespace Multiplayer
         }
     }
 
+    void MultiplayerSystemComponent::SendAllMultiplayerComponentVersionData(AzNetworking::IConnection* connection)
+    {
+        MultiplayerPackets::SyncComponentMismatch componentMismatchPacket;
+        const uint32_t totalMultiplayerComponentCount = aznumeric_cast<uint32_t>(GetMultiplayerComponentRegistry()->GetMultiplayerComponentVersionHashes().size());
+        componentMismatchPacket.SetTotalComponentCount(totalMultiplayerComponentCount);
+
+        for (const auto& multiplayerComponentData : GetMultiplayerComponentRegistry()->GetMultiplayerComponentVersionHashes())
+        {
+            componentMismatchPacket.ModifyComponentVersions().emplace_back(
+                multiplayerComponentData.first, aznumeric_cast<uint64_t>(multiplayerComponentData.second));
+
+            if (componentMismatchPacket.GetComponentVersions().full())
+            {
+                // we've hit the packet limit, send the current packet, and begin filling up another
```

Comment:
This alone makes the "send no manifest" cvar worth it. On a bulky mismatch, it'll keep things quick.

One thing you can consider here is doing one large packet of dynamic size where you store the size of the buffer followed by the buffer. That way you'll leverage AzNetworking's large packet fragmentation. Not strictly necessary but something to think about.

Reply:
sv_versionMismatch_sendAllComponentHashesToClient = false stops sending all information to the client. Think there should be another cvar that eliminates sending all the components to both client and server?

Reply:
Ah, I wasn't aware I could send non-fixed size vectors. I didn't want to assume a large fixed size since there really isn't any cap on the number of components a game can have. Dynamic packet size would let me send just one packet, point me at any existing packet example if you get a chance.

Reply:
I don't think it's been attempted but look at AzContainerSerializers.h. Non-fixed size containers have an implementation along with non-fixed size strings.

The question on sv_versionMismatch_sendAllComponentHashesToClient is an implementation detail for you to consider.

Reply:
Love it! I'm using a dynamically sized map and the code is a lot cleaner. Check out AzContainerSerializers.h, I updated it to include pairs since that's needed for the map iterator, but otherwise it worked without a problem.